workflows/rangeland.nf

```diff
@@ -4,37 +4,156 @@
+    aoi_file       = file( "$params.aoi" )
+    endmember_file = file( "$params.endmember" )
 
     //
-    // MODULE: Run FastQC
+    // MODULE: untar
     //
-    FASTQC (
-        ch_samplesheet
-    )
-    ch_multiqc_files = ch_multiqc_files.mix(FASTQC.out.zip.collect{it[1]})
-    ch_versions = ch_versions.mix(FASTQC.out.versions.first())
+    tar_versions = Channel.empty()
+    if (params.input_tar) {
+        UNTAR_INPUT([[:], params.input])
+        base_path = UNTAR_INPUT.out.untar.map(it -> it[1])
+
+        data = base_path.map(it -> file("$it/*/*", type: 'dir')).flatten()
+        data = data.flatten().filter{ inRegion(it) }
+
+        tar_versions = tar_versions.mix(UNTAR_INPUT.out.versions)
+    } else {
+        data = Channel.fromPath( "${params.input}/*/*", type: 'dir') .flatten()
+        data = data.flatten().filter{ inRegion(it) }
+    }
+
+    if (params.dem_tar) {
+        UNTAR_DEM([[:], params.dem])
+        dem = UNTAR_DEM.out.untar.map(it -> file(it[1]))
+
+        tar_versions = tar_versions.mix(UNTAR_DEM.out.versions)
+    } else {
+        dem = file("$params.dem")
+    }
+
+    if (params.wvdb_tar) {
+        UNTAR_WVDB([[:], params.wvdb])
+        wvdb = UNTAR_WVDB.out.untar.map(it -> file(it[1]))
+
+        tar_versions = tar_versions.mix(UNTAR_WVDB.out.versions)
+    } else {
+        wvdb = file("$params.wvdb")
+    }
+
+
+    //
+    // SUBWORKFLOW: Preprocess satellite imagery
+    //
+    PREPROCESSING(data, dem, wvdb, cube_file, aoi_file)
+    ch_versions = ch_versions.mix(PREPROCESSING.out.versions)
+
+    //
+    // SUBWORKFLOW: Generate trend files and visualization
+    //
+    HIGHER_LEVEL(PREPROCESSING.out.tiles_and_masks, cube_file, endmember_file)
+    ch_versions = ch_versions.mix(HIGHER_LEVEL.out.versions)
+
+    grouped_trend_data = HIGHER_LEVEL.out.mosaic_files.map{ it[1] }.flatten().buffer( size: Integer.MAX_VALUE, remainder: true )
+
+    //
+    // MODULE: Check results
+    //
+    if (params.config_profile_name == 'Test profile') {
+        woody_change_ref      = file("$params.woody_change_ref")
+        woody_yoc_ref         = file("$params.woody_yoc_ref")
+        herbaceous_change_ref = file("$params.herbaceous_change_ref")
+        herbaceous_yoc_ref    = file("$params.herbaceous_yoc_ref")
+        peak_change_ref       = file("$params.peak_change_ref")
+        peak_yoc_ref          = file("$params.peak_yoc_ref")
+
+        CHECK_RESULTS(grouped_trend_data, woody_change_ref, woody_yoc_ref, herbaceous_change_ref, herbaceous_yoc_ref, peak_change_ref, peak_yoc_ref)
+        ch_versions = ch_versions.mix(CHECK_RESULTS.out.versions)
+    }
+
+    if (params.config_profile_name == 'Full test profile') {
+        UNTAR_REF([[:], params.reference])
+        ref_path = UNTAR_REF.out.untar.map(it -> it[1])
+        tar_versions.mix(UNTAR_REF.out.versions)
+
+        CHECK_RESULTS_FULL(grouped_trend_data, ref_path)
+        ch_versions = ch_versions.mix(CHECK_RESULTS_FULL.out.versions)
+    }
```

Comment:
You should not embed test specific code within the pipeline itself, (it's not particularly realistic), for this you should add `nf-test` to the pipeline and use that for a more structured/standardised approach.

A few pipelines now have this (ampliseq, rnaseq, etc.) but if you need pointers let me know.